tests/test_finite_difference.py

```diff
@@ -1,102 +1,57 @@
 import symfit as sf
 import numpy as np
 import pytest
+from tests.auto_variables import *
 
 
 def setup_method():
     np.random.seed(0)
 
 
-def test_1_1_model():
-    '''Tests the case with 1 component and 1 parameter'''
-    x, y = sf.variables('x, y')
-    a = sf.Parameter(name='a')
-    model = sf.Model({y: 3 * a * x**2})
-    x_data = np.arange(10)
-
-    exact = model.eval_jacobian(x=x_data, a=3.5)
-    approx = model.finite_difference(x=x_data, a=3.5)
-    _assert_equal(exact, approx)
-
-    exact = model.eval_jacobian(x=3, a=3.5)
-    approx = model.finite_difference(x=3, a=3.5)
-    _assert_equal(exact, approx)
-
-
-def test_1_multi_model():
-    '''Tests the case with 1 component and multiple parameters'''
-    x, y = sf.variables('x, y')
-    a, b = sf.parameters('a, b')
-    model = sf.Model({y: 3 * a * x**2 - sf.exp(b) * x})
-    x_data = np.arange(10)
-
-    exact = model.eval_jacobian(x=x_data, a=3.5, b=2)
-    approx = model.finite_difference(x=x_data, a=3.5, b=2)
-    _assert_equal(exact, approx)
-    
-    exact = model.eval_jacobian(x=3, a=3.5, b=2)
-    approx = model.finite_difference(x=3, a=3.5, b=2)
-    _assert_equal(exact, approx)
-
-
-def test_multi_1_model():
-    '''Tests the case with multiple components and one parameter'''
-    x, y, z = sf.variables('x, y, z')
-    a, = sf.parameters('a')
-    model = sf.Model({y: 3 * a * x**2,
-                      z: sf.exp(a*x)})
-    x_data = np.arange(10)
-
-    exact = model.eval_jacobian(x=x_data, a=3.5)
-    approx = model.finite_difference(x=x_data, a=3.5)
-    _assert_equal(exact, approx)
-
-    exact = model.eval_jacobian(x=3, a=3.5)
-    approx = model.finite_difference(x=3, a=3.5)
-    _assert_equal(exact, approx)
-
-
-def test_multi_multi_model():
-    '''Tests the case with multiple components and multiple parameters'''
-    x, y, z = sf.variables('x, y, z')
-    a, b, c = sf.parameters('a, b, c')
-    model = sf.Model({y: 3 * a * x**2 + b * x - c,
-                      z: sf.exp(a*x - b) * c})
-    x_data = np.arange(10)
-
-    exact = model.eval_jacobian(x=x_data, a=3.5, b=2, c=5)
-    approx = model.finite_difference(x=x_data, a=3.5, b=2, c=5)
-    _assert_equal(exact, approx, rel=1e-3)
-
-    exact = model.eval_jacobian(x=3, a=3.5, b=2, c=5)
-    approx = model.finite_difference(x=3, a=3.5, b=2, c=5)
-    _assert_equal(exact, approx, rel=1e-3)
-
-
-def test_multi_indep():
+@pytest.mark.parametrize('x_data', [np.arange(10), 3])
+@pytest.mark.parametrize('init_model,init_par',
+    [
+        ({y: 3 * a * x**2}, {'a': 3.5}),
+        ({y: 3 * a * x**2 - sf.exp(b) * x},
+        {'a': 3.5, 'b': 2}),
+        ({y: 3 * a * x**2, z: sf.exp(a*x)}, {'a': 3.5}),
+        ({y: 3 * a * x**2 + b * x - c, z: sf.exp(a*x - b) * c},
+            {'a': 3.5, 'b': 2, 'c': 5})  
```

Comment:
Linting here

Reply:
Because we use global instances of `Variable ` and `Parameter`. I can't think of a way around the problem.

Reply:
The indentation is off in regards to the brackets. You could do something like this for the long ones maybe
```
        (
            model_dict,
            intial_values
        )
```
i.e. take more vertical space to save horizontal space, but indent the values accordingly. Alternatively:
```
        (model_dict,
         intial_values)
```
Note the alignment of the model_dict and initial_values

Reply:
Then I misinterpreted the linting error :D Thank you for the clarification! :)
Just knew it from my cpp compiler and changed it ;)